### PR TITLE
[ADL] ME FIPS Mode Enablement

### PR DESCRIPTION
**Pull request by koktong-ong**
Implement a function to support FIPS mode enablement in ADL

Test: Booted with Windows and Yocto
Verified with FIPS enablement support in ADLN

Signed-off-by: Ong Kok Tong <kok.tong.ong@intel.com>

### CODE REVIEW

File: Silicon/AlderlakePkg/Library/HeciMeExtLib/HeciMeExtLib.c [hunk 1]
```diff
@@ -605,3 +605,117 @@ HeciArbSvnGetInfoMsg (
...
+  if ((GetFipsMode.Response.MkhiHeader.Fields.Command == GEN_GET_FIPS_MODE_CMD) &&
+      (GetFipsMode.Response.MkhiHeader.Fields.IsResponse == 1) &&
+      (GetFipsMode.Response.MkhiHeader.Fields.Result == MkhiStatusSuccess)) {
+    *GetFipsModeData = GetFipsMode.Response.Data;
+  } else {
+    return EFI_DEVICE_ERROR;
+  }
+
+  return Status;
+}
```
Comment (sindhuragrandhi) [inline Silicon/AlderlakePkg/Library/HeciMeExtLib/HeciMeExtLib.c:721]: Add new line

File: Platform/AlderlakeBoardPkg/Library/Stage2BoardInitLib/Stage2BoardInitLib.c [hunk 2]
```diff
@@ -386,8 +387,14 @@ BoardInit (
...
-      SioInit();
+    if (FeatureCfgData != NULL){
+      if (FeatureCfgData->Sio == 1){
+        SioInit();
+      }
+      GetFipsModeData = NULL;
+      HeciGetFipsMode(GetFipsModeData);
+      DEBUG ((DEBUG_INFO, "HeciGetFipsMode = 0x%lx\n", *GetFipsModeData));
+      HeciSetFipsMode(FeatureCfgData->MeFipsMode);
     }
```
Comment (lsubash) [inline Platform/AlderlakeBoardPkg/Library/Stage2BoardInitLib/Stage2BoardInitLib.c:398]: Do HeciSetFipsMode if the state is different.

Comment (lsubash) [inline Platform/AlderlakeBoardPkg/Library/Stage2BoardInitLib/Stage2BoardInitLib.c:398]: Can the FIPS enabled from FIT strap options?

File: Silicon/AlderlakePkg/Include/MkhiMsgsAdl.h [hunk 3]
```diff
@@ -60,28 +60,6 @@
 
 #define ARB_SVN_COMMIT_ALL                      0xFF ///< Value 0xFF refers to committing all pending ARBSVN values.
 
-//
```
Comment (lsubash) [inline Silicon/AlderlakePkg/Include/MkhiMsgsAdl.h:63]: What is the reason for moving these API's to common file?
I would recommend to maintain in ADL. We can review and move these definitions to common later,

File: Platform/AlderlakeBoardPkg/Library/Stage2BoardInitLib/Stage2BoardInitLib.c [hunk 4]
```diff
@@ -386,8 +387,14 @@ BoardInit (
...
       InitializeSmbiosInfo ();
     }
-    if (FeatureCfgData != NULL && FeatureCfgData->Sio == 1){
-      SioInit();
+    if (FeatureCfgData != NULL){
+      if (FeatureCfgData->Sio == 1){
+        SioInit();
+      }
+      GetFipsModeData = NULL;
+      HeciGetFipsMode(GetFipsModeData);
```
[the remark this answers is not in the document]
Comment (koktong-ong) [reply Platform/AlderlakeBoardPkg/Library/Stage2BoardInitLib/Stage2BoardInitLib.c:395]: FIPS mode is not included in current MBP data HOBS

File: Platform/AlderlakeBoardPkg/Library/Stage2BoardInitLib/Stage2BoardInitLib.c (hunk 2 above)
Comment (koktong-ong) [reply Platform/AlderlakeBoardPkg/Library/Stage2BoardInitLib/Stage2BoardInitLib.c:398]: Checked with Kevin James and there is no FIPS related straps settings in MFIT tool

File: Silicon/AlderlakePkg/Include/MkhiMsgsAdl.h (hunk 3 above)
Comment (koktong-ong) [reply Silicon/AlderlakePkg/Include/MkhiMsgsAdl.h:63]: There is MKHI_MESSAGE_HEADER dependencies in common MkhiMsgs header file which causing duplicate definition issue if we merge into Adl specific header file. Currently only the header definition is in common header but the API still in ADL specific Heci lib

File: Silicon/AlderlakePkg/Library/HeciMeExtLib/HeciMeExtLib.c (hunk 1 above)
Comment (koktong-ong) [reply Silicon/AlderlakePkg/Library/HeciMeExtLib/HeciMeExtLib.c]: Revised in latest patch

File: Platform/AlderlakeBoardPkg/Library/Stage2BoardInitLib/Stage2BoardInitLib.c (hunk 2 above)
Comment (koktong-ong) [reply Platform/AlderlakeBoardPkg/Library/Stage2BoardInitLib/Stage2BoardInitLib.c:401]: Revised in latest patch

**Review by lsubash — APPROVED** [review]

File: Silicon/AlderlakePkg/Include/Library/HeciMeExtLib.h [hunk 5]
```diff
@@ -98,4 +99,32 @@ HeciGetHerHashData (
...
+  Send Set FIPS Mode to Enabled or Disabled
+
+  @retval EFI_UNSUPPORTED         Current ME mode doesn't support this function
+  @retval EFI_SUCCESS             Command succeeded
+  @retval EFI_DEVICE_ERROR        HECI Device error, command aborts abnormally
+  @retval EFI_TIMEOUT             HECI does not return the buffer before timeout
+  @retval EFI_BUFFER_TOO_SMALL    Message Buffer is too small for the Acknowledge
+**/
+EFI_STATUS
+HeciSetFipsMode (
```
Comment (gdong1) [inline Silicon/AlderlakePkg/Include/Library/HeciMeExtLib.h:112]: Add "EFIAPI" modifier for the library APIs in both header file and implementation.

**Review by gdong1 — COMMENTED** [review]
it looks good after adding missing "EFIAPI".

**Review by gdong1 — APPROVED** [review]

**Pull request merged**